### PR TITLE
[Batch] `az batch`: Remove deprecated commands and add new parameter `--enable-accelerated-networking` for `az batch pool create` command to enable accelerated networking

### PR DESCRIPTION
**Pull request by JJJessieWang**
**Related command**

az batch

**Description**

- Upgrade the Batch data plane to 14.0.0 (API version 2023-05-01.17.0).
- Upgrade the Batch mgmt plane to 17.1.0 (API version 2023-05-01).
- Add deprecation information for certificate operations.

**Testing Guide**

**History Notes**

[Batch] Remove `az batch job all-statistics` and `az batch pool all-statistics` which no longer worked in the past
[Batch] `az batch pool create`: Add new parameter `--enable-accelerated-networking` to determine whether this pool should enable accelerated networking

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Batch

**Comment by JJJessieWang** [comment]
If I understand correctly, the release is expected at 2023-08-01 as mentioned here. Milestones - Azure/azure-cli (github.com)<https://github.com/Azure/azure-cli/milestones>
Just to confirm. Thanks.

Jessie

File: src/azure-cli/azure/cli/command_modules/batch/commands.py [hunk 1]
```diff
@@ -141,7 +140,6 @@ def get_data_factory(name):
         g.batch_command('delete', 'remove_nodes')
 
     with self.command_group('batch job', get_data_type('job'), client_factory=get_data_factory('job')) as g:
```
Comment (wanghoppe) [inline src/azure-cli/azure/cli/command_modules/batch/commands.py:142]: We can also add the deprecating info to certificate related commands, docs: https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#deprecating-commands-and-arguments

File: src/azure-cli/azure/cli/command_modules/batch/_params.py [hunk 2]
```diff
@@ -243,6 +243,8 @@ def load_arguments(self, _):
         c.argument('target_node_communication_mode', options_list=['--target-communication'],
                    help="The desired node communication mode for the pool. If this element is present, it replaces the existing targetNodeCommunicationMode configured on the Pool. If omitted, any existing metadata is left unchanged.",
                    arg_type=get_enum_type(NodeCommunicationMode))
+        c.extra('enable_accelerated_networking', arg_group="Pool: Network Configuration",
```
[the remark this answers is not in the document]
Comment (JJJessieWang) [reply src/azure-cli/azure/cli/command_modules/batch/_params.py:246]: You're right.

**Comment by JJJessieWang** [comment]
@yanzhudd This PR is ready to review for Azure SDK team.

File: src/azure-cli/azure/cli/command_modules/batch/_params.py [hunk 3]
```diff
@@ -243,6 +243,8 @@ def load_arguments(self, _):
         c.argument('target_node_communication_mode', options_list=['--target-communication'],
                    help="The desired node communication mode for the pool. If this element is present, it replaces the existing targetNodeCommunicationMode configured on the Pool. If omitted, any existing metadata is left unchanged.",
                    arg_type=get_enum_type(NodeCommunicationMode))
+        c.extra('enable_accelerated_networking', options_list=['--accelerated-networking'], arg_group="Pool: Network Configuration",
+                help='Whether this pool should enable accelerated networking. Accelerated networking enables single root I/O virtualization (SR-IOV) to a VM, which may lead to improved networking performance. For more details, see: https://learn.microsoft.com/azure/virtual- network/accelerated-networking-overview. Set true to enable.')
```
Comment (yanzhudd) [inline src/azure-cli/azure/cli/command_modules/batch/_params.py:247]: If the parameter is bool type, consider add `arg_type=get_three_state_flag()` to it.
```suggestion
        c.extra('enable_accelerated_networking', arg_type=get_three_state_flag(), options_list=['--accelerated-networking'], arg_group="Pool: Network Configuration",
                help='Whether this pool should enable accelerated networking. Accelerated networking enables single root I/O virtualization (SR-IOV) to a VM, which may lead to improved networking performance. For more details, see: https://learn.microsoft.com/azure/virtual- network/accelerated-networking-overview. Set true to enable.')
```

**Comment by yanzhudd** [comment]
Hi @JJJessieWang, Azure CLI has bi-annual breaking change releases coinciding with **Microsoft Build and Ignite**, and breaking changes are only be released in the breaking change windows. You could use deprecation process firstly to hide the decrecated commands and parameters ([command deprecation guildline](https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#deprecating-commands-and-arguments)), and release the breaking change in breaking change window. More details could found in the [doc](https://eng.ms/docs/cloud-ai-platform/azure-core/azure-management-and-platforms/control-plane-bburns/azure-cli-tools-azure-cli-powershell-and-terraform/azure-cli-tools/teams_docs/azcli_docs/breaking_changes).

**Comment by JJJessieWang** [comment]
> Hi @JJJessieWang, Azure CLI has bi-annual breaking change releases coinciding with **Microsoft Build and Ignite**, and breaking changes are only be released in the breaking change windows. You could use deprecation process firstly to hide the decrecated commands and parameters ([command deprecation guildline](https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#deprecating-commands-and-arguments)), and release the breaking change in breaking change window. More details could be found in the [doc](https://eng.ms/docs/cloud-ai-platform/azure-core/azure-management-and-platforms/control-plane-bburns/azure-cli-tools-azure-cli-powershell-and-terraform/azure-cli-tools/teams_docs/azcli_docs/breaking_changes).

Hi @yanzhudd, thanks for your professional suggestions. 

- We've deleted this method in Python package. so remain this command and set deprecated flag on it will cause error when customer call. 
- This change has been announced for a long time, suppose all the customer are aware of this change. 
- I didn't see any change whether set hide as True or False on CLI side. 
- When is the next CLI release cycle expected to happen? Sorry that I didn't find information in these links.  

What should I do in this situation?

**Comment by yanzhudd** [comment]
> > Hi @JJJessieWang, Azure CLI has bi-annual breaking change releases coinciding with **Microsoft Build and Ignite**, and breaking changes are only be released in the breaking change windows. You could use deprecation process firstly to hide the decrecated commands and parameters ([command deprecation guildline](https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#deprecating-commands-and-arguments)), and release the breaking change in breaking change window. More details could be found in the [doc](https://eng.ms/docs/cloud-ai-platform/azure-core/azure-management-and-platforms/control-plane-bburns/azure-cli-tools-azure-cli-powershell-and-terraform/azure-cli-tools/teams_docs/azcli_docs/breaking_changes).
> 
> Hi @yanzhudd, thanks for your professional suggestions.
> 
> * We've deleted this method in Python package. so remain this command and set deprecated flag on it will cause error when customer call.
> * This change has been announced for a long time, suppose all the customer are aware of this change.
> * I didn't see any change whether set hide as True or False on CLI side.
> * When is the next CLI release cycle expected to happen? Sorry that I didn't find information in these links.
> 
> What should I do in this situation?

Hi @JJJessieWang 
Do you mean the deprecated commands already cannot work before this change?

**Comment by JJJessieWang** [comment]
> > > Hi @JJJessieWang, Azure CLI has bi-annual breaking change releases coinciding with **Microsoft Build and Ignite**, and breaking changes are only be released in the breaking change windows. You could use deprecation process firstly to hide the decrecated commands and parameters ([command deprecation guildline](https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#deprecating-commands-and-arguments)), and release the breaking change in breaking change window. More details could be found in the [doc](https://eng.ms/docs/cloud-ai-platform/azure-core/azure-management-and-platforms/control-plane-bburns/azure-cli-tools-azure-cli-powershell-and-terraform/azure-cli-tools/teams_docs/azcli_docs/breaking_changes).
> > 
> > 
> > Hi @yanzhudd, thanks for your professional suggestions.
> > 
> > * We've deleted this method in Python package. so remain this command and set deprecated flag on it will cause error when customer call.
> > * This change has been announced for a long time, suppose all the customer are aware of this change.
> > * I didn't see any change whether set hide as True or False on CLI side.
> > * When is the next CLI release cycle expected to happen? Sorry that I didn't find information in these links.
> > 
> > What should I do in this situation?
> 
> Hi @JJJessieWang Do you mean the deprecated commands already cannot work before this change?

Yes. To be specific, it cannot be used from June 2023, when the new version Python package was released (azure-batch==14.0.0). Deprecation of this pool/job lifetime statistics is part of the plan of this release.

**Comment by yanzhudd** [comment]
> > > > Hi @JJJessieWang, Azure CLI has bi-annual breaking change releases coinciding with **Microsoft Build and Ignite**, and breaking changes are only be released in the breaking change windows. You could use deprecation process firstly to hide the decrecated commands and parameters ([command deprecation guildline](https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#deprecating-commands-and-arguments)), and release the breaking change in breaking change window. More details could be found in the [doc](https://eng.ms/docs/cloud-ai-platform/azure-core/azure-management-and-platforms/control-plane-bburns/azure-cli-tools-azure-cli-powershell-and-terraform/azure-cli-tools/teams_docs/azcli_docs/breaking_changes).
> > > 
> > > 
> > > Hi @yanzhudd, thanks for your professional suggestions.
> > > 
> > > * We've deleted this method in Python package. so remain this command and set deprecated flag on it will cause error when customer call.
> > > * This change has been announced for a long time, suppose all the customer are aware of this change.
> > > * I didn't see any change whether set hide as True or False on CLI side.
> > > * When is the next CLI release cycle expected to happen? Sorry that I didn't find information in these links.
> > > 
> > > What should I do in this situation?
> > 
> > 
> > Hi @JJJessieWang Do you mean the deprecated commands already cannot work before this change?
> 
> Yes. To be specific, it cannot be used from June 2023, when the new version Python package was released (azure-batch==14.0.0). Deprecation of this pool/job lifetime statistics is part of the plan of this release.

Hi @JJJessieWang got it. Since it cannot be used already, removing the commands does not seem to be a breaking change. You could remove the commands directly without using deprecation process.

**Review by yanzhudd — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**